codice/ddf
# DDF-04470 Migrating map-context-menu.view.js to React.

#### What does this PR do?

Migrates map-context-menu.view.js to react.

* Corrected logic for showing histogram menu items.
* Fixed React keys error by adding a key to every instance of <React.Fragment> in all of DDF
* Fixed histogram blank display when selecting multiple results.
* Migrated histogram.view to React.
* Migrated inspector.view to React.

#### Who is reviewing it? 
@samuelechu 
@adimka 
@Bdthomson 
@andrewzimmer 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@djblue 

#### How should this be tested?

Before running a search:
1. Right click on the map and verify that both the Histogram and Copy Coordinates options appear
1. Select one of the sub menu options for Copy Coordinates and verify that it works as expected
1. Open the Histogram and verify that it says "Please select a result set to display on the histogram."
1. Search for data a get multiple results.
1. Right click on the map to open the context menu and click Histogram.
1. Verify that Histogram works and shows the expected data.
1. Right click on one of the results in the data.
1. Verify that the menu shows Copy Coordinates, Histogram, and View Inspector with the name of the result you clicked.
1. Shift click at least two results in the data to select multiple results.
1. Right click on the map but not on any results in the data.
1. Verify that the options Copy Coordinates, View Histogram, View Histogram (x selected), and View Inspector (x selected) appear.
1. Right click on one of the results on the map while multiple results are selected.
1. Verify that the options Copy Coordinates, View Histogram, View Histogram (x selected), View Inspector (filename), and View Inspector (x selected) appear.
1. Select View Histogram (x selected) and verify that the histogram only includes the two selected results.
1. Right click again on a result item and select View Inspector (filename) and verify that it shows the inspector for the clicked on result.
1. Right click again and select View Inspector (x selected) and verify that the inspector shows the data for the selected results.

#### Any background context you want to provide?
#### What are the relevant tickets?
Fixes: #4470 

#### Screenshots
<img width="237" alt="screen shot 2019-03-06 at 3 44 16 pm" src="https://user-images.githubusercontent.com/9195522/53912443-eff11580-4026-11e9-8077-bdc92e0d7b20.png">

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
